File: 19_password_strength_checker/python/passwordstrengthchecker.java
password = input("Enter password: ")

has_upper = any(c.isupper() for c in password)
has_lower = any(c.islower() for c in password)
has_digit = any(c.isdigit() for c in password)

score = sum([has_upper, has_lower, has_digit, len(password) >= 8])

if score <= 2:
    print("Weak Password")
elif score == 3:
    print("Medium Password")
else:
    print("Strong Password")

File: 19_password_strength_checker/cpp/password_strength_checker.cpp
#include <iostream>
#include <regex>
using namespace std;

int main() {
    string pwd;
    cout << "Enter password: ";
    cin >> pwd;

    string strength = "Weak";
    if (pwd.length() >= 8 &&
        regex_search(pwd, regex("[A-Z]")) &&
        regex_search(pwd, regex("[a-z]")) &&
        regex_search(pwd, regex("[0-9]")) &&
        regex_search(pwd, regex("[@#$%^&+=]")))
        strength = "Strong";
    else if (pwd.length() >= 6) strength = "Medium";

    cout << "Password strength: " << strength << endl;
}
